spec/models/concerns/check_summable_spec.rb

```diff
@@ -0,0 +1,98 @@
+require 'rails_helper'
+
+RSpec.describe CheckSummable do
+  let(:test_object) { TestClass.new }
+  let(:checksum) { Digest::MD5.new.tap { |digest| digest << 'Test document' }.base64digest }
+  let(:io) { StringIO.new('Test document') }
+
+  before do
+    test_class = Class.new do
+      include CheckSummable
+
+      attr_accessor :as_document_checksum
+
+      def document
+        @document ||= OpenStruct.new(path: 'example.pdf')
+      end
+    end
+
+    stub_const('TestClass', test_class)
+  end
+
+  describe '#calculate_checksum' do
+    subject(:calculate_checksum) { test_object.calculate_checksum(io) }
+
+    context 'with a valid io stream' do
+      it 'returns a base64digest checksum for an IO stream' do
+        expect(calculate_checksum).to eq checksum
+      end
+
+      it 'rewinds the io stream' do
+        expect { calculate_checksum }.not_to change(io, :readline)
+      end
+    end
+
+    context 'with a nil input stream' do
+      let(:io) { nil }
+
+      it 'is nil if the input stream is nil' do
+        expect(calculate_checksum).to be_nil
+      end
+    end
+  end
+
+  describe '#add_checksum' do
+    subject(:add_checksum) { test_object.add_checksum('document') }
+
+    let(:registry) { Paperclip::AdapterRegistry.new }
+
+    before { allow(Paperclip).to receive(:io_adapters).and_return(registry) }
+
+    context 'with a good file' do
+      before do
+        allow(registry).to receive(:for).with(test_object.document).and_return(io)
+      end
+
+      it 'sets the checksum' do
+        expect { add_checksum }.to change(test_object, :as_document_checksum).to checksum
+      end
+    end
+
+    shared_examples 'CheckSummable.add_checksum failure cases' do
```

Comment:
personally don't think we need checksummable here 
```suggestion
    shared_examples 'add_checksum failure cases' do
```